lib/bn.js

```diff
@@ -731,6 +731,44 @@ BN.prototype.uxor = function uxor(num) {
 };
 
 
+// Not ``this`` with ``width`` bitwidth
+BN.prototype.inotn = function inotn(width) {
+  assert(typeof width === 'number' && width >= 0);
+
+  var bytesNeeded = Math.ceil(width / 26) | 0;
+  var bitsLeft = width % 26;
+
+  // Extend the buffer with leading zeroes
+  while (this.length < bytesNeeded) {
+    this.words[this.length++] = 0;
+  }
+
+  if (bitsLeft > 0)
+    bytesNeeded--;
+
+  // Handle complete words
+  for (var i = 0; i < bytesNeeded; i++) {
+    this.words[i] = (this.words[i] & 0xfc000000) |
```

Comment:
I didn't do a full code review to see if anything is done to the upper bits (masked by ```0xfc000000```). I assumed there might be something there and to be on the safe side I decided to keep it as is while NOTing the lower bits.

If you say the top is never used (apart from being an overhead space for in-place operations), that part can be discarded.

IIRC I had weird issues by just doing a NOT on the whole words[i] as obviously that was my first try :)